pkg/storage/watch_cache.go

```diff
@@ -266,7 +266,7 @@ func (w *watchCache) GetAllEventsSinceThreadUnsafe(resourceVersion uint64) ([]wa
 		return result, nil
 	}
 	if resourceVersion < oldest {
-		return nil, errors.NewInternalError(fmt.Errorf("too old resource version: %d (%d)", resourceVersion, oldest))
+		return nil, errors.NewBadRequest(fmt.Sprintf("too old resource version: %d (%d)", resourceVersion, oldest))
```

Comment:
As long as it's not one of the more frequently used return codes I don't have a preference. 404 is what you generally use for things like http://www.google.com/lkajsdlkf, and of course a 5xx means we messed up. I agree 410 is generally what one would apply for: I know for sure this resource used to exist, but it's been deleted, and for it to apply in this scenario the "resource" in question needs to be the in-memory watch window etcd has flushed.